<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PaintWidget</class>
 <widget class="QMainWindow" name="PaintWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>680</width>
    <height>520</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>680</width>
    <height>520</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>680</width>
    <height>520</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>PaintWidget</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(38, 38, 39);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="Painter" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>500</width>
      <height>500</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="maze_page">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QRadioButton {
    background-color:    rgb(38, 38, 39);;
    color:                  white;
}

QRadioButton::indicator {
    width:                  8px;
    height:                 8px;
    border-radius:          6px;
}

QRadioButton::indicator:checked {
    background-color:       black;
    border:                 2px solid white;
}

QRadioButton::indicator:unchecked {
    background-color:       white;
    border:                 2px solid white;
}</string>
    </property>
    <property name="text">
     <string>Maze</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="cave_page">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>10</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QRadioButton {
    background-color:    rgb(38, 38, 39);;
    color:                  white;
}

QRadioButton::indicator {
    width:                  8px;
    height:                 8px;
    border-radius:          6px;
}

QRadioButton::indicator:checked {
    background-color:       black;
    border:                 2px solid white;
}

QRadioButton::indicator:unchecked {
    background-color:       white;
    border:                 2px solid white;
}</string>
    </property>
    <property name="text">
     <string>Cave</string>
    </property>
   </widget>
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>40</y>
      <width>150</width>
      <height>441</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="Maze_page">
     <widget class="QLineEdit" name="text_width_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>70</y>
        <width>90</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">  border: 0px solid gray;
color: #ffffff;</string>
      </property>
      <property name="text">
       <string>Width:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="generate">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>125</y>
        <width>150</width>
        <height>25</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton {
  border: 1px solid gray;
  border-radius: 5px;
  color: #ffffff;
}

QPushButton:hover {
	background-color: #404040;
}

QPushButton:pressed {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #404040, stop: 1 #1a1a1a);
}

</string>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="Cols">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>70</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: #ffffff;</string>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="QPushButton" name="load">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>150</width>
        <height>25</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton {
  border: 1px solid gray;
  border-radius: 5px;
  color: #ffffff;
}

QPushButton:hover {
	background-color: #404040;
}

QPushButton:pressed {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #404040, stop: 1 #1a1a1a);
}

</string>
      </property>
      <property name="text">
       <string>Load from file...</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="text_height_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>95</y>
        <width>90</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">  border: 0px solid gray;
color: #ffffff;</string>
      </property>
      <property name="text">
       <string>Height:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="Rows">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>95</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: #ffffff;</string>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Cave_page">
     <widget class="QSpinBox" name="speed">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>340</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: #ffffff;</string>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::UpDownArrows</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
     </widget>
     <widget class="QPushButton" name="open_file">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>150</width>
        <height>25</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton {
  border: 1px solid gray;
  border-radius: 5px;
  color: #ffffff;
}

QPushButton:hover {
	background-color: #404040;
}

QPushButton:pressed {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #404040, stop: 1 #1a1a1a);
}

</string>
      </property>
      <property name="text">
       <string>Load from file...</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="width">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>70</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: #ffffff;</string>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::UpDownArrows</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="QLineEdit" name="text_chance">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>170</y>
        <width>90</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">  border: 0px solid gray;
color: #ffffff;</string>
      </property>
      <property name="text">
       <string>Initial chance:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="start_button">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>370</y>
        <width>150</width>
        <height>25</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton {
  border: 1px solid gray;
  border-radius: 5px;
  color: #ffffff;
}

QPushButton:hover {
	background-color: #404040;
}

QPushButton:pressed {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #404040, stop: 1 #1a1a1a);
}

</string>
      </property>
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="text_speed">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>340</y>
        <width>90</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">  border: 0px solid gray;
color: #ffffff;</string>
      </property>
      <property name="text">
       <string>Step/msec:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="birth_limit">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>120</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: #ffffff;</string>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::UpDownArrows</enum>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
     </widget>
     <widget class="QLineEdit" name="text_death">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>145</y>
        <width>90</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">  border: 0px solid gray;
color: #ffffff;</string>
      </property>
      <property name="text">
       <string>Death limit:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="death_limit">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>145</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: #ffffff;</string>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::UpDownArrows</enum>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>2</number>
      </property>
     </widget>
     <widget class="QLineEdit" name="text_birth">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>120</y>
        <width>90</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">  border: 0px solid gray;
color: #ffffff;</string>
      </property>
      <property name="text">
       <string>Birth limit:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="step_button">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>305</y>
        <width>150</width>
        <height>25</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton {
  border: 1px solid gray;
  border-radius: 5px;
  color: #ffffff;
}

QPushButton:hover {
	background-color: #404040;
}

QPushButton:pressed {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #404040, stop: 1 #1a1a1a);
}

</string>
      </property>
      <property name="text">
       <string>Next step</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="height">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>95</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: #ffffff;</string>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::UpDownArrows</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="QLineEdit" name="text_height">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>95</y>
        <width>90</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">  border: 0px solid gray;
color: #ffffff;</string>
      </property>
      <property name="text">
       <string>Height:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="text_width">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>70</y>
        <width>90</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">  border: 0px solid gray;
color: #ffffff;</string>
      </property>
      <property name="text">
       <string>Width:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="generate_button">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>200</y>
        <width>150</width>
        <height>25</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton {
  border: 1px solid gray;
  border-radius: 5px;
  color: #ffffff;
}

QPushButton:hover {
	background-color: #404040;
}

QPushButton:pressed {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #404040, stop: 1 #1a1a1a);
}

</string>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="init_chance">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>170</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">color: #ffffff;</string>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::UpDownArrows</enum>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
      <property name="value">
       <double>0.500000000000000</double>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="clear_button">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>485</y>
      <width>150</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
  border: 1px solid gray;
  border-radius: 5px;
  color: #ffffff;
}

QPushButton:hover {
	background-color: #404040;
}

QPushButton:pressed {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #404040, stop: 1 #1a1a1a);
}

</string>
    </property>
    <property name="text">
     <string>Clear</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Painter</class>
   <extends>QWidget</extends>
   <header>painter.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
